tsl/src/switch_ts_user.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * This file and its contents are licensed under the Timescale License.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-TIMESCALE for a copy of the license.
+ */
+#ifndef TIMESCALEDB_TSL_SWITCH_TS_USER_H
+#define TIMESCALEDB_TSL_SWITCH_TS_USER_H
+#include <postgres.h>
+#include <miscadmin.h>
+
+#include "catalog.h"
+#include "extension_constants.h"
+
+/*switch to ts user for _timescaledb_internal access */
+#define SWITCH_TO_TS_USER(schemaname, newuid, saved_uid, saved_secctx)                             \
```

Comment:
these are not part of this PR i just moved them to a separate file

Reply:
Understood, but since you are already refactoring this in terms of moving things around, maybe just make these `static inline` functions?

Or will this touch a lot of other files?

(preferably, this can be done as an initial commit in same PR).

Reply:
if we make them functions there are other places that could be refactored to use them as well making this a slightly bigger change that should not be done as part of this PR